oval_graph/parts/footer.html

```diff
@@ -1,13 +1,137 @@
 </script>
+
 <body>
-    <div id="data"></div>
+	<div id="data"></div>
+	<div id="modal" class="modal">
+		<div class="modal-content">
+			<span id="close" class="close">&times;</span>
+			<div id="content"></div>
+		</div>
+	</div>
 	<script>
-	var data = JSON.parse(JSON.stringify(data_of_tree));
-	$('#data').jstree({
-            'core': {
-                'data': [data]
-            }
-		});	
+		function prepare_file_object(table_object) {
+			var permission = {
+				'uread': null,
+				'uwrite': null,
+				'uexec': null,
+				'gread': null,
+				'gwrite': null,
+				'gexec': null,
+				'oread': null,
+				'owrite': null,
+				'oexec': null
+			}
+			var new_table_object = {};
+			Object.keys(table_object).forEach(key => {
+				if (key in permission) {
+					permission[key] = table_object[key];
+				} else {
+					new_table_object[key] = table_object[key];
+				}
+			});
+			var out = '<code>';
+			Object.keys(permission).forEach(key => {
+				if (permission[key] == 'true') {
+					switch (key.substring(1, key.length)) {
+						case 'read':
+							out += "r";
+							break;
+						case 'write':
+							out += "w";
+							break;
+						case 'exec':
+							out += "x";
+							break;
+						default:
+							// pass
+							break;
+					}
+				} else {
+					out += "-";
```

Comment:
This is not so simple - in OVAL, we may either say that the state is `uread` on or off, which you handle well. But if we don't care, you will visualize it the same as if we want it off. I suggest using e.g. `?` to indicate the "we don't care" option.